wordpress-mobile/WordPress-Login-Flow-Android
# Update `minSdkVersion` (from `21` to `24`)

This PR upgrades Login to `minSdkVersion` to `24`. This has been unblocked because:

- `WPAndroid` was already on `minSdkVersion = 24`, for some time now (see [here](https://github.com/wordpress-mobile/WordPress-Android/pull/15135)).
- `WCAndroid` has been recently upgraded to `minSdkVersion = 24` as well (see [here](https://github.com/woocommerce/woocommerce-android/pull/7643)).

-----

As part of this `minSdkVersion = 24` upgrade the below changes were also applied in order to fix any additional warnings that this change brought-up:

Warnings Resolution List:

1. [Resolve obsolete sdk int lint warning for res values v23.](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/commit/20a7ea4f00f7d9c9f8308e8950f93efc220c00db)
2. [Remove unnecessary duplicate v27 theme login flow config.](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/commit/937f16449688cac7e9296f9cafea9a882abac3e9)

Dependency Config List:

1. [Rename kotlin ktx to androidx core version.](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/commit/b5aa5c8ee38743caba188494c88214dae3556b02)
2. [Extract all dependency version to root build gradle.](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/commit/7d132c4f5349a52f75ba9cc72c6acb2d9286ecf5)
3. [Add androidx prefix to app compat version.](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/commit/6b078f1e60fa069ef6a6c87e475906f739a1caa2)
4. [Move androidx app compat/core and dagger versions to main group.](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/commit/2507d767794c3896e8f7189c697485b2a2598752)

Dependency Update List:
1. [Update robolectric to 4.9.](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/commit/1a862eb14e7bb1c9d9932a1bc49fa4f3a6c7c235)

-----

PS: @ovitrif @atorresveiga I added you as the main reviewers, that is, in addition to the @wordpress-mobile/apps-infrastructure team itself, but randomly, since I just wanted someone from both, the `WordPress` and `Woo Commerce` teams, to sign-off on that change for WPAndroid and WCAndroid alike.

-----

### Testing instructions

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough.
- However, if you want to be thorough about reviewing this change, you could:
    - Quickly smoke test, either the `WordPress` or `Woo Commerce` apps, with this version of `Login`, and see if it works as expected.
    - Try and install, either the `WordPress` or `Woo Commerce` apps, with this version of `Login`, on Android 5 (`API 21/22`) and Android 6 (`API 23`) devices and verify that you can't (see [API Levels](https://apilevels.com/)).
